### PR TITLE
Fix throw `AstroUserError` in mdx

### PR DESCRIPTION
**Pull request by liruifengv**
## Changes

- Fix throw astroUserError in mdx

```astro
---
import { AstroError } from 'astro/errors';

throw new AstroError('Error message', 'Error hint not display');
---

<div>
Throw a custom AstroUserError in mdx.
</div>
```

Use this compoent in mdx, above code will throw a custom AstroUserError, but error.hint is not display, because it was catched by `throwEnhancedErrorIfMdxComponent` function.

- see: https://stackblitz.com/edit/github-f2fmuc?file=src%2Fcontent%2Fblog%2Fusing-mdx.mdx

### CODE REVIEW

**Comment by liruifengv** [comment]
Related Starlight PR: https://github.com/withastro/starlight/pull/1542, Starlight use `AstroError`

File: .changeset/brave-pots-listen.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+throw original AstroUserError in mdx
```
Comment (ematipico) [inline .changeset/brave-pots-listen.md:5]:
```suggestion
Improves DX by throwing the original `AstroUserError` when an error is thrown inside a `.mdx` file.
```

**Review by ematipico — APPROVED** [review]
I proposed a suggestion for the changeset, let me know what you think

**Comment by liruifengv** [comment]
> I proposed a suggestion for the changeset, let me know what you think

Thank you

**Pull request merged**